app/models/warehouse/case_report.rb

```diff
@@ -0,0 +1,161 @@
+        case_report.casework_officer_user_id = kase.decorate.casework_officer_user&.id
+        case_report.business_group_id = kase.responding_team&.business_group&.id
+        case_report.directorate_id = kase.responding_team&.directorate&.id
+        case_report.director_general_name_property_id = kase.responding_team&.business_group&.properties&.lead&.singular_or_nil&.id # Director General name
+        case_report.director_name_property_id = kase.responding_team&.directorate&.properties&.lead&.singular_or_nil&.id # Director name
+        case_report.deputy_director_name_property_id = kase.responding_team&.properties&.lead&.singular_or_nil&.id # Deputy Director name
+
+        # Report fields - for output
+        case_report.number = kase.number
+        case_report.case_type = kase.decorate.pretty_type
+        case_report.current_state = I18n.t("helpers.label.state_selector.#{kase.current_state}")
+        case_report.responding_team = kase.responding_team&.name
+        case_report.responder = kase.responder&.full_name
+        case_report.date_received = kase.received_date
+        case_report.internal_deadline = kase.flagged? ? kase.internal_deadline : nil
+        case_report.external_deadline = kase.external_deadline
+        case_report.date_responded = kase.date_responded
+        case_report.date_compliant_draft_uploaded = kase.date_draft_compliant
+        case_report.trigger = kase.flagged? ? 'Yes' : nil
+        case_report.name = kase.name
+        case_report.requester_type = kase.sar? ? nil : kase.requester_type.humanize
+        case_report.message = self.dequote_and_truncate(kase.message)
+        case_report.info_held = kase.info_held_status&.name
+        case_report.outcome = kase.outcome&.name
+        case_report.refusal_reason = kase.refusal_reason&.name
+        case_report.exemptions = kase.exemptions.map{ |x| CaseClosure::Exemption.section_number_from_id(x.abbreviation) }.join(',')
+        case_report.postal_address = kase.postal_address
+        case_report.email = kase.email
+        case_report.appeal_outcome = kase.appeal_outcome&.name
+        case_report.third_party = kase.respond_to?(:third_party) ? self.humanize_boolean(kase.third_party) : nil
+        case_report.reply_method = kase.respond_to?(:reply_method) ? kase.reply_method.humanize : nil
+        case_report.sar_subject_type = kase.respond_to?(:subject_type) ? kase.subject_type.humanize : nil
+        case_report.sar_subject_full_name = kase.respond_to?(:subject_full_name) ? kase.subject_full_name : nil
+        case_report.business_unit_responsible_for_late_response = kase.decorate.late_team_name
+        case_report.extended = extension_count(kase) > 0 ? 'Yes' : 'No'
+        case_report.extension_count = self.extension_count(kase)
+        case_report.deletion_reason = kase.reason_for_deletion
+        case_report.casework_officer = kase.casework_officer
+        case_report.created_by = kase.creator.full_name
```

Comment:
Can kase.creator ever be nil?

Reply:
In theory no because all records prior to March 2019 have the 'default user' assigned as the creator (id: -100) with the name as an empty string.